sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -93,5 +94,31 @@ func (b *BearerTokenPolicy) Do(req *policy.Request) (*http.Response, error) {
 			}
 		}
 	}
-	return res, err
+	return res, ensureNonRetriable(err)
+}
+
+func ensureNonRetriable(err error) error {
+	if err != nil {
+		if _, ok := err.(errorinfo.NonRetriable); !ok {
```

Comment:
I think you should use `errors.As()` into an `errorinfo.NonRetriable` just in case `err` needs to be `Unwrap()`'ed.